scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingMethodImpl.java

```diff
@@ -195,6 +196,25 @@ public int schedule() {
         return tasksStarted;
     }
 
+    private void updateStatsAboutPendingEligibleAndNeededNodes() {
+        updateStatsAboutPendingEligibleAndNeededNodes(Collections.EMPTY_LIST);
+    }
+
+    private void updateStatsAboutPendingEligibleAndNeededNodes(Collection<? extends TaskDescriptor> taskDescriptors) {
+        // Pending eligible
+        final int pendingEligible = taskDescriptors.size();
+
+        schedulingService.getListener().updatePendingEligibleTasks(pendingEligible);
+
+        // Needed nodes
+        final int nodesNeeded = taskDescriptors.stream().mapToInt(TaskDescriptor::getNumberOfNodesNeeded).sum();
+
+        getRMProxiesManager().getRmProxy().setPendingTasksCount(nodesNeeded);
```

Comment:
Could you please clarify why you set pending tasks count to nodes Needed amount?